src/zmalloc.c

```diff
@@ -715,6 +767,90 @@ int zmalloc_get_allocator_info(size_t *allocated, size_t *active, size_t *reside
     return 1;
 }
 
+/* Compute the total memory wasted in fragmentation of inside small arena bins
+ * in the specified arena. Done by summing the memory in unused regs in all
+ * slabs of all small bins. */
+size_t zmalloc_get_frag_smallbins_by_arena(unsigned int arena) {
```

Comment:
modified directly on zmalloc_get_frag_smallbins()? the only differences are MALLCTL_ARENAS_ALL and arena.